kpi/deployment_backends/kobocat_backend.py

```diff
@@ -682,6 +684,16 @@ def get_attachment_objects_from_dict(self, submission: dict) -> QuerySet:
             instance_id=submission['_id']
         )
 
+    def get_daily_counts(self, timeframe: tuple[date, date]) -> dict:
+        daily_counts = ReadOnlyKobocatDailyXFormSubmissionCounter.objects.values(
+            'date', 'counter'
+        ).filter(
+            xform_id=self.xform_id,
+            date__range=timeframe,
+        )
+        data = {str(count['date']): count['counter'] for count in daily_counts}
+        return data
```

Comment:
Hmmm.. We will need to handle the users with partial permissions. 
Let's discuss about it.

Reply:
As per our discussion, users with partial submissions cannot user daily counters. 
We are going to use `MongoDB.aggregate()` to calculate the correct value on the fly. Not as fast as daily submission counters but after few tests on production servers, it is pretty fast thanks to indexes.